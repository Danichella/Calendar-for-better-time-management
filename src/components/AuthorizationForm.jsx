import React from 'react';
import '../assets/sass/authorization.scss';
import Logo from '../assets/img/Logo.png';

export const AuthorizationForm = ({
  data,
  onChangeHandler,
  submitTitle,
  onSubmitClick,
  children,
}) => {
  return (
    <div className="authorization-wrapper">
      <div className="authorization-form-wrapper">
        <img src={Logo} alt="Logo" className="authorization-form-logo" />
        {data.map((element, index) => (
          <div key={index}>
            <p className="authorization-input-label authorization-space-between">
              {`${element.title[0].toUpperCase()}${element.title.slice(1)}:`}
            </p>
            <input
              type={element.type ?? 'text'}
              className="authorization-input"
              placeholder={element.placeholder ?? `Enter your ${element.title}`}
              value={element.value ?? ''}
              onChange={(e) => onChangeHandler(element.id, e.target.value)}
            />
          </div>
        ))}
        <button
          className="authorization-submit-button authorization-space-between"
          onClick={onSubmitClick}
        >
          {submitTitle}
        </button>
        <p className="authorization-additional-message authorization-space-between">
          {children}
        </p>
      </div>
    </div>
  );
};
