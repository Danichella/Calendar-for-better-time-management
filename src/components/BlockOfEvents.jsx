import React from 'react';
import { useNavigate } from 'react-router-dom';
import '../assets/sass/timetable.scss';

export const BlockOfEvents = ({ eventsList, currentMonth, currentDay }) => {
  const navigate = useNavigate();
  const matchedEvents = eventsList?.filter((event) => {
    const date = new Date(event.date);
    return date.getMonth() === currentMonth && date.getDate() === currentDay;
  });

  if (!matchedEvents || matchedEvents.length === 0) return <></>;

  return (
    <div>
      {matchedEvents.map((event, index) => (
        <div
          key={index}
          className="timetable-day-of-month-event"
          onClick={() => navigate(`/events/${event.id}`)}
        >
          {event.title}
        </div>
      ))}
    </div>
  );
};
