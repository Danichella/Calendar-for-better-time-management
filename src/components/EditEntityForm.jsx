import '../assets/sass/authorization.scss';

export const EditEntityForm = ({
  data,
  onChangeHandler,
  helperText = null,
}) => {
  return (
    <div className="authorization-wrapper">
      <div className="authorization-form-wrapper">
        {data.map((element, index) => (
          <div key={index}>
            <p className="authorization-input-label authorization-space-between">
              {`${element.title[0].toUpperCase()}${element.title.slice(1)}:`}
            </p>
            <input
              type={element.type ?? 'text'}
              className="authorization-input"
              placeholder={element.placeholder ?? `Enter your ${element.title}`}
              value={element.value ?? ''}
              onChange={(e) => onChangeHandler(element.id, e.target.value)}
            />
          </div>
        ))}
        {helperText && (
          <p className="authorization-helper-text">{helperText}</p>
        )}
      </div>
    </div>
  );
};
