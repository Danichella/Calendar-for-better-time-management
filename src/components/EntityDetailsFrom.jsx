import '../assets/sass/entity_details.scss';
import '../assets/sass/authorization.scss';
import EditIcon from '../assets/img/EditIcon.svg';

export const EntityDetailsForm = ({ entityType, data, onEditClick }) => {
  return (
    <div className="entity-details-wrapper">
      <div className="authorization-form-wrapper">
        <div className="entity-details-header-wrapper">
          <p>{`${entityType} Details`}</p>
          <img src={EditIcon} alt="Edit" onClick={onEditClick} />
        </div>
        {data.map((element, index) =>
          Array.isArray(element.value) ? (
            <div
              key={index}
              className="entity-details-data-field-with-multiple-values-wrapper"
            >
              <p>Invited users:</p>
              {element.value.map((value, index) => (
                <p key={index}>{value}</p>
              ))}
            </div>
          ) : (
            <div key={index} className="entity-details-data-field-wrapper">
              <p>{element.title}</p>
              <p>{element.value}</p>
            </div>
          )
        )}
      </div>
    </div>
  );
};
