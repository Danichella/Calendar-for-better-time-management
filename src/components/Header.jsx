import '../assets/sass/header.scss';
import Logo from '../assets/img/Logo.png';

export const Header = ({ secondary, primary }) => {
  return (
    <div className="header-wrapper">
      <img src={Logo} alt="Logo" className="header-logo" />
      <div className="header-actions-wrapper">
        <button
          className="header-actions-secondary-button"
          onClick={secondary.callback}
        >
          {secondary.title}
        </button>
        <button
          className="header-actions-primary-button"
          onClick={primary.callback}
        >
          {primary.title}
        </button>
      </div>
    </div>
  );
};
