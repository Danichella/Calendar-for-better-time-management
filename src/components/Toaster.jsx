import '../assets/sass/toaster.scss';
import SuccessIcon from '../assets/img/SuccessIcon.svg';
import ErrorIcon from '../assets/img/ErrorIcon.svg';
import ReactDOM from 'react-dom';

export const addNotification = (type, message) => {
  const toast = (
    <div className={`toaster-element toaster-element-${type}`}>
      <img src={type === 'success' ? SuccessIcon : ErrorIcon} />
      <p>{message}</p>
    </div>
  );

  ReactDOM.render(toast, document.getElementById('toaster'));
  setTimeout(() => {
    document
      .getElementById('toaster')
      .removeChild(document.getElementById('toaster').childNodes[0]);
  }, 3000);
};
