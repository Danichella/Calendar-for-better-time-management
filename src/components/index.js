export * from './Header';
export * from './Timetable';
export * from './EntityDetailsFrom';
export * from './EditEntityForm';
export * from './AuthorizationForm';
export * from './BlockOfEvents';
export * from './Toaster';
