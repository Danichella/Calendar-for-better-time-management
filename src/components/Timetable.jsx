import { useState, useEffect } from 'react';
import '../assets/sass/timetable.scss';
import ArrowLeftIcon from '../assets/img/ArrowLeftIcon.svg';
import ArrowRightIcon from '../assets/img/ArrowRightIcon.svg';
import { FormatArrayOfDays } from '../helpers';
import { BlockOfEvents } from './BlockOfEvents';

const monthTitles = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December',
];

export const Timetable = ({ eventsList }) => {
  const date = new Date();
  const [currentMonth, setCurrentMonth] = useState(date.getMonth());
  const [daysOfMonth, setDaysOfMonth] = useState(
    FormatArrayOfDays(currentMonth)
  );

  const onPrevMonthHandleClick = () => {
    currentMonth === 0
      ? setCurrentMonth(11)
      : setCurrentMonth(currentMonth - 1);
  };
  const onNextMonthHandleClick = () => {
    currentMonth === 11
      ? setCurrentMonth(0)
      : setCurrentMonth(currentMonth + 1);
  };

  useEffect(() => {
    setDaysOfMonth(FormatArrayOfDays(currentMonth));
  }, [currentMonth]);

  return (
    <div>
      <p className="timetable-title">Your Timetable</p>
      <div className="timetable-calendar-actions-wrapper">
        <div className="timetable-month-controller-wrapper">
          <img
            src={ArrowLeftIcon}
            alt="prev"
            onClick={() => onPrevMonthHandleClick()}
          />
          <p>{monthTitles[currentMonth]}</p>
          <img
            src={ArrowRightIcon}
            alt="next"
            onClick={() => onNextMonthHandleClick()}
          />
        </div>
        <button onClick={() => setCurrentMonth(date.getMonth())}>Today</button>
      </div>
      <div className="timetable-grid">
        <div className="timetable-day-of-week-wrapper">Mon</div>
        <div className="timetable-day-of-week-wrapper">Tue</div>
        <div className="timetable-day-of-week-wrapper">Wed</div>
        <div className="timetable-day-of-week-wrapper">Thu</div>
        <div className="timetable-day-of-week-wrapper">Fri</div>
        <div className="timetable-day-of-week-wrapper">Sat</div>
        <div className="timetable-day-of-week-wrapper">Sun</div>
        {daysOfMonth.map((element, index) => {
          let titleOfBlock = (
            <p className="timetable-day-of-month-title">{element.day}</p>
          );

          if (element.type === 'inactive')
            titleOfBlock = (
              <p className="timetable-day-of-month-title timetable-day-of-month-title-inactive">
                {element.day}
              </p>
            );

          if (
            element.type === 'active' &&
            date.getDate() === element.day &&
            date.getMonth() === currentMonth
          )
            titleOfBlock = (
              <p className="timetable-day-of-month-title timetable-today-pointer">
                {element.day}
              </p>
            );

          return (
            <div key={index} className="timetable-day-of-month-wrapper">
              {titleOfBlock}
              {element.type !== 'inactive' && (
                <BlockOfEvents
                  eventsList={eventsList}
                  currentMonth={currentMonth}
                  currentDay={element.day}
                />
              )}
            </div>
          );
        })}
      </div>
    </div>
  );
};
