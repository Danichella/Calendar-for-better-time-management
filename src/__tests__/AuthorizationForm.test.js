import React from 'react';
import { shallow } from 'enzyme';
import { AuthorizationForm } from '../components/AuthorizationForm';

describe('AuthorizationForm', () => {
  let component;
  let onSubmitCallback;
  let onChangeHandler;
  const initialValues = [
    { id: 'username', title: 'username', value: null },
    { id: 'firstName', title: 'first name', value: null },
    { id: 'lastName', title: 'last name', value: null },
    { id: 'email', title: 'email', value: null },
    { id: 'password', title: 'password', value: null, type: 'password' },
    {
      id: 'confirmPassword',
      title: 'confirm password',
      value: null,
      type: 'password',
      placeholder: 'Enter your password again',
    },
  ];

  beforeEach(() => {
    onSubmitCallback = jest.fn();
    onChangeHandler = jest.fn();
    component = shallow(
      <AuthorizationForm
        data={initialValues}
        onChangeHandler={onChangeHandler}
        submitTitle="Sign up"
        onSubmitClick={onSubmitCallback}
      />
    );
  });

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call callback function when on submit button click', () => {
    component.find('.authorization-submit-button').simulate('click');

    expect(onSubmitCallback).toHaveBeenCalledTimes(1);
  });

  it('should call callback function when change input value', () => {
    component
      .find('.authorization-input')
      .first()
      .simulate('change', { target: { value: 'name' } });

    expect(onChangeHandler).toHaveBeenCalledTimes(1);
  });
});
