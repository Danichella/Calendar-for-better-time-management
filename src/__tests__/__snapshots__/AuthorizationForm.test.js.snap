// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorizationForm should match snapshot 1`] = `
<div
  className="authorization-wrapper"
>
  <div
    className="authorization-form-wrapper"
  >
    <img
      alt="Logo"
      className="authorization-form-logo"
      src={
        Object {
          "createTransformer": [Function],
        }
      }
    />
    <div
      key="0"
    >
      <p
        className="authorization-input-label authorization-space-between"
      >
        Username:
      </p>
      <input
        className="authorization-input"
        onChange={[Function]}
        placeholder="Enter your username"
        type="text"
        value=""
      />
    </div>
    <div
      key="1"
    >
      <p
        className="authorization-input-label authorization-space-between"
      >
        First name:
      </p>
      <input
        className="authorization-input"
        onChange={[Function]}
        placeholder="Enter your first name"
        type="text"
        value=""
      />
    </div>
    <div
      key="2"
    >
      <p
        className="authorization-input-label authorization-space-between"
      >
        Last name:
      </p>
      <input
        className="authorization-input"
        onChange={[Function]}
        placeholder="Enter your last name"
        type="text"
        value=""
      />
    </div>
    <div
      key="3"
    >
      <p
        className="authorization-input-label authorization-space-between"
      >
        Email:
      </p>
      <input
        className="authorization-input"
        onChange={[Function]}
        placeholder="Enter your email"
        type="text"
        value=""
      />
    </div>
    <div
      key="4"
    >
      <p
        className="authorization-input-label authorization-space-between"
      >
        Password:
      </p>
      <input
        className="authorization-input"
        onChange={[Function]}
        placeholder="Enter your password"
        type="password"
        value=""
      />
    </div>
    <div
      key="5"
    >
      <p
        className="authorization-input-label authorization-space-between"
      >
        Confirm password:
      </p>
      <input
        className="authorization-input"
        onChange={[Function]}
        placeholder="Enter your password again"
        type="password"
        value=""
      />
    </div>
    <button
      className="authorization-submit-button authorization-space-between"
      onClick={[MockFunction]}
    >
      Sign up
    </button>
    <p
      className="authorization-additional-message authorization-space-between"
    />
  </div>
</div>
`;
